### PR TITLE
fix: moving / removing validations out of property mapper transformations

### LINKED ISSUE #25549: Move propertymapper validation logic out of the transformation logic where possible
### Description

#24811 introduced a generic validation method to property mappers that can be used more generally for validation, rather than the current logic that is in the transformation methods.

A follow-on to #25434 could benefit from this because the why that runtime properties are validated is different than when it was triggered at build time

### PR DESCRIPTION
**Pull request by shawkins**
closes: #25549

The rationale for this change is that it moves up in time when the error occurs - to be just like any other property / usage error. The only logic that was not conducive to move was HttpPropertyMappers.getHttpEnabledTransformer given the usage of the ConfigSourceInterceptorContext - but there should be a way to do it without that.

cc @mabartos @vmuzikar

### CODE REVIEW

**Comment by mabartos** [comment]
> The only logic that was not conducive to move was HttpPropertyMappers.getHttpEnabledTransformer given the usage of the ConfigSourceInterceptorContext - but there should be a way to do it without that.

Changes LGTM, but could you elaborate a little bit?

**Comment by shawkins** [comment]
> @shawkins Changes LGTM, but could you elaborate a little bit?

I didn't try it, but we should just be able to make a general call into Configuration.get... rather than proceeding on the context.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java [hunk 1]
```diff
@@ -29,6 +30,7 @@ public abstract class AbstractStartCommand extends AbstractCommand implements Ru
     public void run() {
         doBeforeRun();
         CommandLine cmd = spec.commandLine();
+        HttpPropertyMappers.validateConfig();
```
Comment (mabartos) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java:33]: I don't really like to have it here, but as we do not have any better solution ATM, I'm ok with these changes for now. 
However, we'd need to consider a better way how to handle these validations (even for the import/export validations) in a more abstract way. I mean, a rational question might be: "Why do the httpPropertyMappers need to be validated in the abstract start command?". 

We should leverage the `PropertyMapper.validator()` as much as possible for the individual mappers.
Or we could introduce some validation mechanism for the whole property mappers classes. I mean, having general validations for the whole option category or sth like that.

It'd be good to have sth like:
```java
interface PropertyMapperWrapper { // or some more suitable name 
    Collection<PropertyMapper<?>> getPropertyMappers();

    default Collection<PropertyMapperValidator> getValidators(){
        return Collections.emptyList();
    }

    default boolean isEnabled() {
        return true;
    }
}

interface PropertyMapperValidator{
    void validate();  
}
```

However, it's out of the scope of this PR, and a follow-up task for it should be created. 
@shawkins WDYT?
cc @vmuzikar

**Review by mabartos — APPROVED** [review]
@shawkins I like the removal of the `InitializationException` and the parent issue should be properly resolved by this PR. 

I've just added a comment below. I don't consider it as a blocker and definitely out of the scope of this PR, but it's very closely related. 

Approved as changes LGTM with considerations of some new follow-up tasks.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java (hunk 1 above)
Comment (shawkins) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java:33]: @mabartos let's move this onto #27025 - I had just updated that one to say it probably wasn't worth persuing, but if you are onboard with introducing something, then let's leave it open.

Comment (mabartos) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java:33]: Ok, let's move it there. As I've mentioned, it's ok for me to directly use the static method for the validation for now.

**Comment by mabartos** [comment]
@vmuzikar As you are mentioned here, do you wanna review it as well?

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java (hunk 1 above)
Comment (vmuzikar) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractStartCommand.java:33]: @mabartos +1. I don’t think the validations belong here either but I'm ok with it if we move it in a follow-up.

**Review by vmuzikar — APPROVED** [review]
@shawkins LGTM, thanks.

**Pull request merged**